cmd/describe/ingress/cmd.go

```diff
@@ -0,0 +1,185 @@
+	"github.com/openshift/rosa/pkg/rosa"
+)
+
+// Regular expression to used to make sure that the identifier given by the
+// user is safe and that it there is no risk of SQL injection:
+var ingressKeyRE = regexp.MustCompile(`^[a-z0-9]{3,5}$`)
+
+var Cmd = &cobra.Command{
+	Use:     "ingress",
+	Short:   "Show details of the specified ingress within cluster",
+	Example: `rosa describe ingress <ingress_id> -c mycluster`,
+	Run:     run,
+	Args: func(_ *cobra.Command, argv []string) error {
+		if len(argv) != 1 {
+			return fmt.Errorf(
+				"Expected exactly one command line parameter containing the id of the ingress",
+			)
+		}
+		return nil
+	},
+}
+
+func init() {
+	ocm.AddClusterFlag(Cmd)
+	output.AddFlag(Cmd)
+}
+
+func run(_ *cobra.Command, argv []string) {
+	r := rosa.NewRuntime().WithOCM()
+	defer r.Cleanup()
+
+	ingressID := argv[0]
+	if !ingressKeyRE.MatchString(ingressID) {
+		r.Reporter.Errorf(
+			"Ingress  identifier '%s' isn't valid: it must contain only letters or digits",
+			ingressID,
+		)
+		os.Exit(1)
+	}
+
+	clusterKey := r.GetClusterKey()
+	cluster := r.FetchCluster()
+
+	// Try to find the ingress:
+	r.Reporter.Debugf("Loading ingresses for cluster '%s'", clusterKey)
+	ingresses, err := r.OCMClient.GetIngresses(cluster.ID())
+	if err != nil {
+		r.Reporter.Errorf("Failed to get ingresses for cluster '%s': %v", clusterKey, err)
+		os.Exit(1)
+	}
+
+	var ingress *cmv1.Ingress
+	for _, item := range ingresses {
+		if ingressID == "apps" && item.Default() {
+			ingress = item
+		}
+		if ingressID == "apps2" && !item.Default() {
+			ingress = item
+		}
+		if item.ID() == ingressID {
+			ingress = item
+		}
+	}
+	if ingress == nil {
+		r.Reporter.Errorf("Failed to get ingress '%s' for cluster '%s'", ingressID, clusterKey)
+		os.Exit(1)
+	}
+	if output.HasFlag() {
+		var b bytes.Buffer
+		err := cmv1.MarshalIngress(ingress, &b)
+		if err != nil {
+			r.Reporter.Errorf("Failed to generate output for ingress '%s': %v", ingress.ID(), err)
+			os.Exit(1)
+		}
+		ret := make(map[string]interface{})
+		err = json.Unmarshal(b.Bytes(), &ret)
+		if err != nil {
+			r.Reporter.Errorf("Failed to generate output for ingress '%s': %v", ingress.ID(), err)
+			os.Exit(1)
+		}
+		err = output.Print(ret)
+		if err != nil {
+			r.Reporter.Errorf("Failed to output ingress '%s': %v", ingress.ID(), err)
+			os.Exit(1)
+		}
+		return
+	}
+	entries := generateEntriesOutput(cluster, ingress)
+	ingressOutput := ""
+	keys := helper.MapKeys(entries)
+	sort.Strings(keys)
+	minWidth := getMinWidth(keys)
+	for _, key := range keys {
+		ingressOutput += fmt.Sprintf("%s: %s\n", key, strings.Repeat(" ", minWidth-len(key))+entries[key])
+	}
+	fmt.Print(ingressOutput)
+}
+
+func getMinWidth(keys []string) int {
+	minWidth := 0
+	for _, key := range keys {
+		if len(key) > minWidth {
+			minWidth = len(key)
+		}
+	}
+	return minWidth
+}
+
+func generateEntriesOutput(cluster *cmv1.Cluster, ingress *cmv1.Ingress) map[string]string {
```

Comment:
@gdbranco Can we can some test coverage on this function please.

cheers,
Rob